src/GraphQL/Execution/DefaultExecutionStrategySelector.cs

```diff
@@ -0,0 +1,80 @@
+using GraphQLParser.AST;
+
+namespace GraphQL.Execution
+{
+    /// <inheritdoc cref="IExecutionStrategySelector"/>
+    public class DefaultExecutionStrategySelector : IExecutionStrategySelector
+    {
+        private readonly ExecutionStrategyRegistration[] _registrations;
+
+        /// <summary>
+        /// Initializes an instance that only returns the default registrations;
+        /// <see cref="ParallelExecutionStrategy"/> for <see cref="OperationType.Query"/> and
+        /// <see cref="SerialExecutionStrategy"/> for <see cref="OperationType.Mutation"/>.
+        /// </summary>
+        public DefaultExecutionStrategySelector()
+            : this(Array.Empty<ExecutionStrategyRegistration>())
+        {
+        }
+
+        /// <summary>
+        /// Initializes an instance that returns the specified registrations.
+        /// </summary>
+        public DefaultExecutionStrategySelector(IExecutionStrategy queryExecutionStrategy, IExecutionStrategy mutationExecutionStrategy, IExecutionStrategy? subscriptionExecutionStrategy = null)
+            : this(GetRegistrations(queryExecutionStrategy, mutationExecutionStrategy, subscriptionExecutionStrategy))
+        {
+        }
```

Comment:
Could eliminate this.  Seems handy to manually select the execution strategy.  However, if an execution strategy is registered within the DI engine as an `IExecutionStrategy` (our code would not do this), the DI engine would select this constructor because it fulfills the most number of constructor arguments.

Reply:
Per your comments, I will remove this constructor.

Reply:
Removed.  Lmk if you want it restored or resolve if not.